spec/services/teacher_training_public_api/sync_all_providers_and_courses_spec.rb

```diff
@@ -3,18 +3,42 @@
 RSpec.describe TeacherTrainingPublicAPI::SyncAllProvidersAndCourses, sidekiq: true do
   include TeacherTrainingPublicAPIHelper
 
-  before do
-    allow(described_class).to receive(:sync_providers)
-  end
-
   describe '.call' do
     context 'paginates the correct number of pages' do
+      before do
+        allow(described_class).to receive(:sync_providers)
+      end
+
       it 'calls sync providers 3 times' do
         stub_teacher_training_api_providers_with_multiple_pages
         described_class.call
 
         expect(described_class).to have_received(:sync_providers).exactly(3).times
       end
     end
+
+    context 'a previous year recruitment cycle' do
+      let(:recruitment_cycle_year) { RecruitmentCycle.previous_year }
+      let(:sync_provider) { instance_double(TeacherTrainingPublicAPI::SyncProvider) }
+
+      before do
+        allow(sync_provider).to receive(:call)
+      end
+
+      it 'calls sync provider with the previous year recruitment cycle' do
+        stub_teacher_training_api_providers_with_single_page(recruitment_cycle_year: recruitment_cycle_year)
+
+        allow(TeacherTrainingPublicAPI::SyncProvider)
+            .to receive(:new)
+            .with(provider_from_api: anything, recruitment_cycle_year: recruitment_cycle_year)
+            .and_return(sync_provider)
+
+        described_class.call(recruitment_cycle_year: recruitment_cycle_year)
+
+        expect(TeacherTrainingPublicAPI::SyncProvider)
+            .to have_received(:new)
+            .with(provider_from_api: anything, recruitment_cycle_year: recruitment_cycle_year)
```

Comment:
I think we want to be testing that our instance_double `sync_provider` has its `call` method called.

I'd put the setup in the before block, so that `SyncProvider` is set up to return `sync_provider`. Then the spec can just be: Call the described_class call method -> expect sync_provider to have received call. It's probably fine to not explicitly expect the `new method to be called with the correct recruitment cycle, since the test will break if it isn't.

Reply:
Makes sense, we can put that in

Reply:
Cool, changes have been made